packages/core/src/core/fetch.ts

```diff
@@ -292,25 +311,70 @@ const fetchAndProcessMergeErrors = async (
   try {
     const progressReporters = _.mapValues(
       adapters,
-      (_adapter, adapterName) => createAdapterProgressReporter(adapterName, 'fetch', progressEmitter)
+      (_adapter, accountName) => createAdapterProgressReporter(accountName, 'fetch', progressEmitter)
     )
     const fetchResults = await Promise.all(
       Object.entries(adapters)
-        .map(async ([adapterName, adapter]) => {
+        .map(async ([accountName, adapter]) => {
           const fetchResult = await adapter.fetch({
-            progressReporter: progressReporters[adapterName],
+            progressReporter: progressReporters[accountName],
           })
+          const { updatedConfig, errors } = fetchResult
+          if (fetchResult.elements.length > 0 && accountName !== fetchResult.elements[0]
+            .elemID.adapter) {
```

Comment:
nit: line breaks in the middle of expressions make it harder to read - maybe 
```suggestion
          if (
            fetchResult.elements.length > 0
            && accountName !== fetchResult.elements[0].elemID.adapter
          ) {
```
or
```suggestion
          if (fetchResult.elements.length > 0
              && accountName !== fetchResult.elements[0].elemID.adapter) {
```